Category to pathway type

MonikaFu opened the pull request:
This PR updates the codebase to refer to Pathway Type (`pathway_type`), rather than Category (`category`), as part of adapting to the finalized data model

MonikaFu committed:
Change category to pathway_type

MonikaFu committed:
Style

MonikaFu committed:
Use pathway_type instead of category

AlexAxthelm committed:
Merge branch 'main' into category-to-pathway-type

AlexAxthelm committed:
Update tests for SearchSection

AlexAxthelm committed:
Update badge tests to use pathway_type instead

AlexAxthelm committed:
Update tests for ScenarioCard

Use `pathway_type` instead of `category`

AlexAxthelm committed:
`prettier` formatting

AlexAxthelm committed:
Remove last code vestiges of `category`

AlexAxthelm committed:
respond to Copilot review

Update R code import documentation, and update comment for section block

jdhoffa commented on src/components/SearchSection.test.tsx:
```diff
@@ -34,8 +34,8 @@ vi.mock("../data/scenariosData", () => ({
       description: "A scenario based on current implemented policies.",
-      category: "Forecast",
-      category_tooltip: "Forecast scenarios are based on existing trends.",
+      pathway_type: "Forecast",
```
Agreed. I'm also curious why this doesn't cause our "schema validation" action to fail on this PR? Might be something to handle in #217

AlexAxthelm replied:
I think that's because right now we only validate the fields against the schema as the source files (`.json`, not as objects that are being read in). That is, we have schema validation as a separate process unrelated to build/test

jdhoffa replied:
Interesting... so does validating against the `.json` mean we are only able to check, for example, that the `pathway_type` key exists, but not that the values of it satisfy the enum requirement? 

In any case, something that we should handle more comprehensively in #217 , including a few tests that the validation fails when inappropriate data are used

jdhoffa commented on src/components/SearchSection.test.tsx:
```diff
@@ -159,9 +159,12 @@ describe("SearchSection", () => {
+    //expect(screen.getByText("Exploration")).toBeInTheDocument();
+    //expect(screen.getByText("Normative")).toBeInTheDocument();
+    //expect(screen.getByText("Policy")).toBeInTheDocument();
+    //expect(screen.getByText("Projection")).toBeInTheDocument();
```
We should tackle all TODOs before merging to `main`. (Or remove them and open an issue if they really have nothing to do with this PR... though it seems that they are related).

AlexAxthelm replied:
Note that the dropdown only loads what's in the data (no options that yield no results). Add one of each type to the testing data

jdhoffa replied:
New issue created here: https://github.com/RMI/pbtar/issues/234

jdhoffa commented on src/data/README.md:
```diff
@@ -16,8 +16,8 @@ An example scenario metadata JSON file looks like...
-    "category": "IAM",
-    "category_tooltip": "Integrated Assessment Model",
+    "pathway_type": "Policy",
+    "pathway_type_tooltip": "Integrated Assessment Model",
```
Agree

jdhoffa commented on src/data/scenarios_metadata_1.json:
```diff
@@ -3,8 +3,8 @@
-    "category": "IAM",
-    "category_tooltip": "Integrated Assessment Model",
+    "pathway_type": "Projection",
+    "pathway_type_tooltip": "Projection of future developments in sectors.",
```
NB (and not strictly related to this PR): It seems _very_ brittle to define the `pathway_type_tooltip` manually, when it is uniquely determined by whatever the `pathway_type` is. I would prefer centrally managing a mapping from `pathway_type` to `pathway_type_tooltip` (likely, somewhere close to the `pbtar_schema`) and then programatically generating that

AlexAxthelm replied:
I think that's a holdover from it being `category`, and not having an finite set of values. That is, it might have arbitrary abbreviations, which would need tooltipped for more info.

jdhoffa replied:
Tracked in #235

jdhoffa reviewed (changes requested):
See comments

AlexAxthelm committed:
Merge branch 'main' into category-to-pathway-type

AlexAxthelm committed:
Update testing and documentation fields

Update `pathway_type_tooltip`s in docs to realistic fields, and testing
JSON object to one that could pass schema check.

AlexAxthelm committed:
test: ensure that pathway_types in search dropdown

AlexAxthelm committed:
refactor: use `pathway_type` instead of `category`

Following the finalization of the data model, the `category` field was
reworked into `pathway_type`

empty commit, to wrap up the PR

AlexAxthelm committed:
Merge branch 'main' into category-to-pathway-type

AlexAxthelm commented:
@jdhoffa [5fd3706](https://github.com/RMI/pbtar/pull/227/commits/5fd370634ab5fd47e316f5d1607d9d735d2de35e) closes #234. Do we want those filters to appear when there aren't any results for them?

jdhoffa commented:
> @jdhoffa [5fd3706](https://github.com/RMI/pbtar/pull/227/commits/5fd370634ab5fd47e316f5d1607d9d735d2de35e) closes #234. Do we want those filters to appear when there aren't any results for them?

Nope, this looks great, thank you.
I had opened #234 before GitHub was loading 5fd3706 for me for some reason. Happy to close it without action

jdhoffa reviewed (approved):
Looks great!!
